### DISCUSSION TITLE
浏览器现在可以默认切换到火狐上上嘛

### DISCUSSION THREAD

**Discussion by Arondight** [Arondight/Adachi-BOT#756] [Q&A] [answered]
1. 部署难度极大降低
2. 莫名其妙的内存问题有可能会缓解或加重

目前我还不知道怎么让 puppeter 用火狐……

**Comment by Arondight** [comment, marked as answer]
哦，不对，改一下变量值的路径就行了，但是这样有个问题，就是 puppeteer 用了系统火狐，用户就没法用了……我应该在某一天试过。另外目前换火狐反而会增加部署难度，因为puppeteer 也下载不了火狐，一样被墙了

**Comment by mark9804** [comment]
如果用国内的 apt/yum/pacman 源的话有可能正常下载吗，比如 tuna

**Comment by mark9804** [comment]
纯技术角度来说 puppeteer 是能用 firefox 的，直接在他们 readme 里面搜索 firefox 就可以，他们会自己下载 firefox nightly，应该不会跟系统火狐冲突才对？我没试过

**Reply by Arondight** [reply]
不搞了，还是直接使用包管装浏览器部署方便

**Comment by mark9804** [comment]
刚刚发现 firefox 对于 preload 参数的处理方式好像跟 chromium 有点不一样（上 firefox，下 chromium）我觉得可能是 `crossorigin` 到底要不要加的问题，暂时先以 chromium 为准
<img width="490" alt="image" src="https://user-images.githubusercontent.com/9006264/161361452-7c6c89e7-9af2-4a3b-b8b8-591e72c48cac.png">
<img width="490" alt="image" src="https://user-images.githubusercontent.com/9006264/161361462-f2e743fe-4e16-4144-b45c-ae93c7675567.png">
